editor/src/components/canvas/canvas-strategies/strategies/absolute-resize-bounding-box-strategy.spec.browser2.tsx

```diff
@@ -364,6 +407,462 @@ describe('Absolute Resize Strategy', () => {
       `),
     )
   })
+  it('resizes absolute element with aspect ratio locked, keeping aspect ratio', async () => {
+    const renderResult = await renderTestEditorWithCode(
+      projectWithAspectRatioLockedElement,
+      'await-first-dom-report',
+    )
+
+    const target = EP.elementPath([['0cd', '7a0']])
+    const dragDelta = windowPoint({ x: 29, y: -23 })
+
+    await renderResult.dispatch([selectComponents([target], false)], true)
+    resizeElement(renderResult, dragDelta, EdgePositionTopLeft, emptyModifiers)
+    await renderResult.getDispatchFollowUpActionsFinished()
+
+    expect(getPrintedUiJsCode(renderResult.getEditorState()))
+      .toEqual(`import * as React from 'react'
```

Comment:
I don't like something about these tests: 
because they share the same starter project, the output is very noisy. in these tests you are basically asserting 1-2-4 lines of change, but you hide those lines in 45 lines of project content. 
plus, because the project code is declared up top, you have to scroll a lot if you want to compare the project contents before-and-after. so for tests at the bottom, you scroll up 800 lines, memorize the project contents, scroll to the test, try to find the one line of change.

I think this would be better off if each test would have its own tailored starting project contents, even if that makes the test file longer. for test cases that only need a single element, you could use a projectContents with only that one element, making the diff much easier to see.

Reply:
would it be OK to create a "template" that can be instantiated with differenct top/left/width/height values (such as https://github.com/concrete-utopia/utopia/blob/f887383496127e184284320648bd1b9cf4a5807f/editor/src/components/canvas/canvas-strategies/strategies/absolute-resize-bounding-box-strategy.spec.browser2.tsx#L258)?

Reply:
yeah that would be awesome!

Reply:
You can cut the template down quite a lot too by using `makeTestProjectCodeWithSnippet`

Reply:
There's a good example of this actually that @ruggi introduced: https://github.com/concrete-utopia/utopia/blob/d39e5225c50a7359eff05cbb2e83316fa5aa3515/editor/src/components/canvas/canvas-strategies/strategies/flex-resize-basic-strategy.spec.browser2.tsx#L708-L761

Reply:
I followed Balazs's suggestion and created minimal test fixtures